cmd/github-ldap-user-group-creator/main.go

```diff
@@ -142,12 +148,46 @@ func main() {
 		logrus.WithError(err).Fatal("Failed to load the mapping")
 	}
 
-	if err := ensureGroups(ctx, clients, mapping, opts.dryRun); err != nil {
+	data, err := ioutil.ReadFile(opts.groupsFile)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to read the group file")
+	}
+	roverGroups := map[string][]string{}
```

Comment:
Using a list of strings is prone to e.g. containing duplicate names and other mistakes (empty string is a string too). I'd add at least basic validation of the input:

- duplicate items
- both group names and member names should be checked to be valid names

Reply:
Done.